src/js/services/app-service.js

```diff
@@ -0,0 +1,34 @@
+import { store } from 'stores/app-store';
+import * as AppStateActions from 'actions/app-state-actions';
+import { observable } from 'utils/events';
+
+
+export function openWidget() {
+    let {embedded} = store.getState().appState;
+    if (!embedded) {
+        store.dispatch(AppStateActions.openWidget());
+        observable.trigger('widget:opened');
+    }
+}
+
+export function closeWidget() {
+    let {embedded} = store.getState().appState;
+    if (!embedded) {
+        store.dispatch(AppStateActions.closeWidget());
+        observable.trigger('widget:closed');
+    }
+}
+
+
+export function toggleWidget() {
+    let {embedded, widgetOpened} = store.getState().appState;
+    if (!embedded) {
+        store.dispatch(AppStateActions.toggleWidget());
```

Comment:
ohhh I thought you were talking about the action creators... my bad.. you are right, we call those instead.